src/util/curves.ts

```diff
@@ -0,0 +1,136 @@
+
+// Factorial
+function fact(n: number): number {
+  let x = 1;
+  for (let i = 2; i <= n; i++) {
+    x *= i;
+  }
+  return x;
+}
+
+// Binomial Coefficients
+function bincoeff(n: number, k: number): number {
+  if (k == 0) {
+    return 1;
+  }
+  return fact(n) / (fact(k) * (fact(n - k)));
+}
+
+function range(n: number): number[] {
+  return [...Array(n)].map((item: any, index: number) => index);
+}
+
+// calculate bezier coefficients: binomial coefficients
+function bezier_coeff(n: number): number[] {
+  return range(n + 1).map((i: number) => bincoeff(n, i));
```

Comment:
AFAICT this is basically computing a row of Pascal's triangle. Doing it this way with repeated computations of binomial coefficients is not very efficient compared to the dynamic programming approach but this is probably fine.